pkg/api/store/cluster/cluster_store.go

```diff
@@ -633,6 +639,38 @@ func setBackupConfigSecretKeyIfNotExists(oldData, newData map[string]interface{}
 	}
 }
 
+func validateS3Credentials(data map[string]interface{}) error {
+	s3BackupConfig := values.GetValueN(data, "rancherKubernetesEngineConfig", "services", "etcd", "backupConfig", "s3BackupConfig")
+	if s3BackupConfig == nil {
+		return nil
+	}
+	configMap := convert.ToMapInterface(s3BackupConfig)
+	sbc := &v3.S3BackupConfig{
+		AccessKey: convert.ToString(configMap["accessKey"]),
+		SecretKey: convert.ToString(configMap["secretKey"]),
+		Endpoint:  convert.ToString(configMap["endpoint"]),
+		Region:    convert.ToString(configMap["region"]),
+		CustomCA:  convert.ToString(configMap["customCa"]),
+	}
+
+	s3Client, err := etcdbackup.GetS3Client(sbc)
+	if err != nil {
+		return err
+	}
+	bucket := convert.ToString(configMap["bucketName"])
```

Comment:
is this bucketName already present in configMap, we should do this test before getting client if so